projects/baby_driver/test/test_gpio_set.c

```diff
@@ -0,0 +1,117 @@
+#include "gpio_set.h"
+
+#include <string.h>
+
+#include "babydriver_msg_defs.h"
+#include "can.h"
+#include "can_msg_defs.h"
+#include "can_transmit.h"
+#include "dispatcher.h"
+#include "gpio.h"
+#include "log.h"
+#include "ms_test_helper_can.h"
+#include "ms_test_helpers.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+typedef enum {
+  TEST_CAN_EVENT_TX = 0,
+  TEST_CAN_EVENT_RX,
+  TEST_CAN_EVENT_FAULT,
+  NUM_TEST_CAN_EVENTS,
+} TestCanEvent;
+
+static CanStorage s_can_storage;
+static uint8_t s_times_callback_called;
+static uint8_t s_received_data[8];
+static void *s_received_context;
+
+static StatusCode s_status_return;
+
+static StatusCode prv_rx_gpio_set_callback(uint8_t data[8], void *context, bool *tx_result) {
+  s_times_callback_called++;
+  memcpy(s_received_data, data, 8);
+  s_received_context = context;
+  *tx_result = false;  // avoid TXing result of the status message
+  return s_status_return;
+}
+
+void setup_test(void) {
+  initialize_can_and_dependencies(&s_can_storage, SYSTEM_CAN_DEVICE_BABYDRIVER, TEST_CAN_EVENT_TX,
+                                  TEST_CAN_EVENT_RX, TEST_CAN_EVENT_FAULT);
+  TEST_ASSERT_OK(dispatcher_init());
+  TEST_ASSERT_OK(gpio_set_init());
+  s_times_callback_called = 0;
+  memset(s_received_data, 0, sizeof(s_received_data));
+  s_status_return = STATUS_CODE_OK;
+}
+
+void teardown_test(void) {}
+
+// Test that we can succesfully set the gpio pin to specified value (0 to 1) with an OK status
+void test_setting_gpio(void) {
+  // register a callback for the status message to check the resulting status
+  TEST_ASSERT_OK(
+      dispatcher_register_callback(BABYDRIVER_MESSAGE_STATUS, prv_rx_gpio_set_callback, NULL));

```

Comment:
You should do this in `setup_test` since you do it at the beginning of every test.